File: src/Frontend/monoton_client/lib/Models/Dto/Request/SendMessage.dart
import 'package:monoton_client/Models/Enum/MessageType.dart';
import 'package:json_annotation/json_annotation.dart';
part 'SendMessage.g.dart';

@JsonSerializable()
class SendMessage {
  @JsonKey(name: 'Content')
  List<int> content = []; // Dart uses List<int> for byte arrays
  @JsonKey(name: 'TargetChatId')
  late String targetChatId;
  @JsonKey(name: 'MessageType')
  late MessageType messageType ; // You'll need to define the MessageType enum
  @JsonKey(name: 'ResourceAddress')
  late String resourceAddress ;
  @JsonKey(name: 'IsReplay')
  late bool isReplay;
  @JsonKey(name: 'Metadata')
  late String metadata ;
  SendMessage();
  factory SendMessage.fromJson(Map<String, dynamic> json) => _$SendMessageFromJson(json);
  Map<String, dynamic> toJson() => _$SendMessageToJson(this);
}

File: src/Backend/Monotone/Models/Dto/Request/SendMessage.cs
using System.Text.Json.Serialization;
using Monotone.Models.Enums;
using Monotone.Utilities;

namespace Monotone.Models.Dto.Request;

public class SendMessage
{
    [JsonConverter(typeof(ByteArrayConverter))]
    public byte[] Content { get; set; } = Array.Empty<byte>();
    public string TargetChatId { get; set; } = null!;
    public MessageType MessageType { get; set; }
    public string ResourceAddress { get; set; } = string.Empty;
    public bool IsReplay { get; set; }
    public string Metadata { get; set; } = string.Empty;

}